### DISCUSSION TITLE
Select

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#239] [Component suggestions]
<img width="480" alt="visuele weergave van het select component" src="https://github.com/nl-design-system/backlog/assets/545207/d3711d8a-e0ea-4840-ba36-7d96eae720e0">

We willen graag meer leren over het Select component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1712847916533489)

---

## Naam
Select

## Doel
Biedt de mogelijkheid om één optie te selecteren uit een opengeklapte lijst.

## Ook bekend als
Dropdown

## Gerelateerde componenten
[Select Combobox](https://github.com/orgs/nl-design-system/discussions/340), [Radio Group](https://github.com/orgs/nl-design-system/discussions/222)

## Onderzoek

[`<select>`: The HTML Select element - MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select)
Documentatie van Mozilla Developers Network over het `<select>` HTML element.

[Alice Bartlett: Bin your select tags - Youtube](https://www.youtube.com/watch?v=-dH6a6eMdXE)
Alice Bartlett vertelt waarom het Select component voor problemen kan zorgen. Dit doet zij op basis van gebruikersonderzoek voor GOV.UK. Daarnaast biedt ze alternatieven aan.

[Under-Engineered Select Menus - Adrian Roselli](https://adrianroselli.com/2021/03/under-engineered-select-menus.html)

[Multiselect - Rian Rietveld - Youtube](https://youtu.be/wLpEgj0FWv0)
Rian laat zien waarom het gebruik van een multiselect voor problemen kan zorgen.

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-forms-select--docs)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/select)
- [Figma](https://www.figma.com/file/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?type=design&node-id=4442-22796&mode=design&t=e8Q45bpsxzbtppbV-4)

---

### Screenshots / Varianten

**Figma**
![Scherm­afbeelding 2024-04-16 om 15 30 56](https://github.com/nl-design-system/backlog/assets/545207/0559fe8c-f038-4a33-a498-6591d744e636)

---

### Notities
- Text input component wordt hergebruikt hierdoor Empty, Placeholder en Filled states.
- Icoon: arrow down (blauw)
- Lijkt een custom menu te activeren.

[Bekijk het Text input component van Amsterdam voor meer informatie](https://github.com/orgs/nl-design-system/discussions/225#discussioncomment-8743963)

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-select--docs)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/Select)
- [Figma - WIP](https://www.figma.com/file/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?type=design&node-id=1130-3653&mode=design&t=BtGepOhJNYlAEr1Y-4)
- [Github](https://github.com/nl-design-system/denhaag/tree/main/components/Select)

---

### Screenshots / Varianten

**Storybook**
![Scherm­afbeelding 2024-09-17 om 10 51 26](https://github.com/user-attachments/assets/e97f62b8-877b-417f-8c3a-e08213a38dc6)


**Figma - WIP**
![Scherm­afbeelding 2024-04-16 om 15 39 12](https://github.com/nl-design-system/backlog/assets/545207/d38f382f-2e47-4711-89b5-e786d8f75873)

---

### Notities
- Storybook
  - Icoon: arrow down.
  - Triggert native menu.
- Figma (WIP)
  - Icoon: chevron down
  - Zou een het native menu van de browser triggeren.
In Storybook is ook een [Select van Open Formulieren beschikbaar](https://nl-design-system.github.io/denhaag/?path=/story/open-forms-select--default). Deze werkt net even anders. Met custom menu en optie om deze lijst te filteren door te typen (combobox?).
  
 [Bekijk het Text input component van Den Haag voor meer informatie](https://github.com/orgs/nl-design-system/discussions/225#discussioncomment-8744057)

**Comment by jeffreylauwers** [comment]
## DSO Toolkit

[Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/story/html-css-form-groups-select--select&viewMode=story)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-04-16 om 15 59 02](https://github.com/nl-design-system/backlog/assets/545207/81faa2df-df0e-4c24-8ad4-034df908aea8)

Active
![Scherm­afbeelding 2024-04-16 om 15 59 09](https://github.com/nl-design-system/backlog/assets/545207/65f8e0b3-6d32-4ee7-91fb-756b136b887a)

Multiple = True
![Scherm­afbeelding 2024-04-16 om 15 59 34](https://github.com/nl-design-system/backlog/assets/545207/2c127a31-511a-472f-a47d-86b28b370db4)

---

### Notities
- Ondergebracht bij 'Form group' waardoor Form field label, Form field description, Form field error message, etc. direct worden meegenomen.
- Icoon: arrow down
- Native menu van browser.
- Multiple property biedt de mogelijkheid om er een multiselect van te maken.

[Bekijk het Text input component van DSO Toolkit voor meer informatie](https://github.com/orgs/nl-design-system/discussions/225#discussioncomment-8744225)

**Comment by jeffreylauwers** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/select)

---

### Screenshots / Varianten

**Documentatie website**

Default
![Scherm­afbeelding 2024-04-16 om 16 07 06](https://github.com/nl-design-system/backlog/assets/545207/f1d09e6e-4740-4b25-8c42-dd5a8ba9be56)

Active
![Scherm­afbeelding 2024-04-16 om 16 07 12](https://github.com/nl-design-system/backlog/assets/545207/a4030c22-1955-4bdb-abce-13a148879cc7)

Met status
![Scherm­afbeelding 2024-04-16 om 16 07 22](https://github.com/nl-design-system/backlog/assets/545207/8ad6fe9d-78b5-4f6b-a604-c468b9fba42a)

---

### Notities
- Richtlijnen beschikbaar. Hierin wordt ook expliciet benoemd eerst andere oplossingen te overwegen.
- Icoon: arrow down
- Native menu van browser.

[Bekijk het Text input component van DUO voor meer informatie](https://github.com/orgs/nl-design-system/discussions/225#discussioncomment-8744362)

**Comment by jeffreylauwers** [comment]
## Open Formulieren

- [Storybook- Select](https://open-formulieren.github.io/open-forms-sdk/?path=/story/form-io-components-vanilla-select--select)
- [Storybook - Selectfield](https://open-formulieren.github.io/open-forms-sdk/?path=/docs/pure-react-components-forms-selectfield--docs)

---

### Screenshots / Varianten

**Storybook - Select**

Default
![Scherm­afbeelding 2024-04-16 om 16 17 46](https://github.com/nl-design-system/backlog/assets/545207/17490459-4f5f-41c7-8b74-142d6cd0bcc5)

Active
![Scherm­afbeelding 2024-04-16 om 16 17 51](https://github.com/nl-design-system/backlog/assets/545207/a4045778-8cec-4fe8-b048-9fd8fd694d8a)

![Scherm­afbeelding 2024-04-16 om 16 18 07](https://github.com/nl-design-system/backlog/assets/545207/8357a4cc-3d49-43ec-a782-92c964ea99fa)

Selected
![Scherm­afbeelding 2024-04-16 om 16 17 58](https://github.com/nl-design-system/backlog/assets/545207/0241478f-d86c-4c0b-a1b1-c6a1a793e2fb)

**Storybook - Selectfield**

Default
![Scherm­afbeelding 2024-04-16 om 16 19 53](https://github.com/nl-design-system/backlog/assets/545207/46ea2e0d-9932-4ebd-89ef-af2da28208ac)

Active
![Scherm­afbeelding 2024-04-16 om 16 19 59](https://github.com/nl-design-system/backlog/assets/545207/cb2783e2-892f-4538-97e1-2ce75a328e1f)

Selected
![Scherm­afbeelding 2024-04-16 om 16 20 04](https://github.com/nl-design-system/backlog/assets/545207/54c764ea-e23a-481a-823e-9515083a458b)

Invalid
![Scherm­afbeelding 2024-04-16 om 16 20 29](https://github.com/nl-design-system/backlog/assets/545207/b7c5e40a-d071-4a82-a259-fc265a0ca0be)

---

### Notities
- Twee componenten Select en Selectfield
- Select
  - Custom menu.
  - Icoon: arrow down (arrow up bij active)
  - Biedt de mogelijkheid om te selecteren en binnen de lijst te zoeken (combo box).
  - Selectie wordt getoond als badge, rechts wordt een verwijder optie getoond.
- Selectfield
  - Custom menu.
  - Icoon: chevron down
  - Mogelijkheid voor een AsyncSelectField variant.

[Bekijk het Text input component van Open Formulieren voor meer informatie](https://github.com/orgs/nl-design-system/discussions/225#discussioncomment-8744459)

**Comment by jeffreylauwers** [comment]
## Open Gemeenten

[Patternlab - Forms and Search](https://patternlab.lenteveld.nl/patterns/01-molecules-080-Forms-And-Search/index.html)

---

### Screenshots / Varianten

**Patternlab**

![Scherm­afbeelding 2024-04-16 om 16 26 45](https://github.com/nl-design-system/backlog/assets/545207/759b0bfc-1e01-4dd8-87ed-9fedbb5ada3d)

Active
![Scherm­afbeelding 2024-04-16 om 16 26 53](https://github.com/nl-design-system/backlog/assets/545207/8e45af92-84b2-49f0-a0dc-a51a86319f2e)

---

### Notities
- Onder de naam: SingleSelect (ook een MultiSelect beschikbaar)
- Icoon: arrow down (links een border).
- Native menu van browser.

[Bekijk het Text input component van OpenGemeenten voor meer informatie](https://github.com/orgs/nl-design-system/discussions/225#discussioncomment-8746808)

**Comment by jeffreylauwers** [comment]
## Rotterdam

- [Figma](https://www.figma.com/file/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?type=design&node-id=17363-11238&mode=design&t=7o9engAv5dJyPwE9-4)
- [GitHub](https://github.com/nl-design-system/rotterdam/tree/main/rotterdam-nlds-parent-wicket/rotterdam-nlds-wicket/src/main/java/nl/rotterdam/nl_design_system/wicket/components/select)

---

### Screenshots / Varianten

**Figma**

![Scherm­afbeelding 2024-04-16 om 17 09 06](https://github.com/nl-design-system/backlog/assets/545207/f2f5b724-b767-4444-9e9b-085aa82caf0e)

![Scherm­afbeelding 2024-04-16 om 17 09 10](https://github.com/nl-design-system/backlog/assets/545207/6151ca13-6af9-415e-8e6b-b255b3276624)

---

### Notities
- Genaamd: Dropdown.
- Lijkt een custom menu te activeren.
- Icoon: arrows up/down
- Multi select variant beschikbaar.

[Bekijk het Text input component van Rotterdam voor meer informatie](https://github.com/orgs/nl-design-system/discussions/225#discussioncomment-8747081)

**Comment by jeffreylauwers** [comment]
## RVO

- [Documentatie website](https://nl-design-system.github.io/rvo/docs/components/form-select/)
- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-select--documentatie)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/form-select)
- [Figma](https://www.figma.com/file/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?type=design&node-id=4234-19984&mode=design&t=SLdlq6cK627mWSiH-4)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-04-16 om 17 12 57](https://github.com/nl-design-system/backlog/assets/545207/2a9bcb84-17ec-4fbd-827f-4f91811e930a)

Active
![Scherm­afbeelding 2024-04-16 om 17 13 03](https://github.com/nl-design-system/backlog/assets/545207/e9e6fe04-bd8b-4070-90dd-3be90650d7d4)

**Figma**
![Scherm­afbeelding 2024-04-16 om 17 13 16](https://github.com/nl-design-system/backlog/assets/545207/bf5f6677-d738-4ad9-b112-ce944f9cf02d)

---

### Notities
- Native menu van browser.
- Icoon: chevron down (op gekleurd vlak).

[Bekijk het Text input component van RVO voor meer informatie](https://github.com/orgs/nl-design-system/discussions/225#discussioncomment-8747275)

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-select--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/select)
- [Figma](https://www.figma.com/file/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?type=design&node-id=1130-3653&mode=design&t=O2bp6qVK1n2KREe3-4)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-04-16 om 17 20 37](https://github.com/nl-design-system/backlog/assets/545207/a0c7edd1-5c8f-4c72-9df0-441778c06f57)

Active
![Scherm­afbeelding 2024-04-16 om 17 20 43](https://github.com/nl-design-system/backlog/assets/545207/45fca5b1-4b66-4c10-926f-a6f53c8f565d)

Disabled
![Scherm­afbeelding 2024-04-16 om 17 20 47](https://github.com/nl-design-system/backlog/assets/545207/d2fd7f93-7f7c-43d2-8bb7-61e7c601838b)

Invalid
![Scherm­afbeelding 2024-04-16 om 17 20 51](https://github.com/nl-design-system/backlog/assets/545207/5109b38c-e589-412a-8133-57978d1f3680)

Read-only
![Scherm­afbeelding 2024-04-16 om 17 20 56](https://github.com/nl-design-system/backlog/assets/545207/5d63496d-dcc1-40e7-b93f-80772a1b211d)

Busy
![Scherm­afbeelding 2024-04-16 om 17 21 01](https://github.com/nl-design-system/backlog/assets/545207/f7846c04-0784-4782-8725-3ebb0f8a7cf1)

Focus
![Scherm­afbeelding 2024-04-16 om 17 21 05](https://github.com/nl-design-system/backlog/assets/545207/fdd65d85-a635-42ec-a8d0-daf71d37e2fb)

Focus-visible
![Scherm­afbeelding 2024-04-16 om 17 21 09](https://github.com/nl-design-system/backlog/assets/545207/38f40bf2-5930-4b94-8d7f-cd1d2e96ca77)

Multiple
![Scherm­afbeelding 2024-04-16 om 17 21 14](https://github.com/nl-design-system/backlog/assets/545207/9c3e37aa-d208-4eea-9448-bb12c0cdfb98)

**Figma**

States
![Scherm­afbeelding 2024-04-16 om 17 23 21](https://github.com/nl-design-system/backlog/assets/545207/3557c58b-8986-4917-bd48-204940639174)

---

### Notities
- Icoon: arrow down (links een border).
- Native menu van browser.
- Ook bij Urecht is de [Open Formulieren Select](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/open-forms-select--docs) beschikbaar.

[Bekijk het Text input component van Utrecht voor meer informatie](https://github.com/orgs/nl-design-system/discussions/225#discussioncomment-8747812)

**Comment by MariekeBrouwer** [comment]
Het component gebruikt Groningen voornamelijk in formulieren 
![image](https://github.com/nl-design-system/backlog/assets/149144374/b0cb945d-7964-4dce-b46a-1ed0ee472cd2)
Voor keuzeselectie maken wij onderscheid tussen radio button, checkbox en select. Radio button voor één keuzemogelijkheid met een beperkt aantal opties (ja/nee/misschien of appel/peer/banaan). Checkbox voor 'ik ga akkoord met' en meerkeuze, Select wanneer er één keuzemogelijkheid is en de lijst wat groter is. 

![image](https://github.com/nl-design-system/backlog/assets/149144374/1105ce30-a93a-4d65-b869-e2670c2948ec)
Ook bij filteren op pagina's wordt het select component gebruikt.

Oh en de functionaliteit van de select komt bijvoorbeeld ook terug in de zoekbalk:
![image](https://github.com/nl-design-system/backlog/assets/149144374/d2a2708b-8c6e-4f04-8807-7b2df79b83a0)

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Select op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=select)

**Comment by Robbert** [comment]
Ik vraag me af: hoe gaan screen readers om met een lege default select, zoals: de eerste `option` hier:

```html
<select>
  <option value=""></option>
  <option value="AMS">Amsterdam</option>
</select>
```

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Select op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=select+%2334)

**Comment by jeffreylauwers** [comment]
Eerder had ik voor het Voorbeeld thema de token `todo.select.icon.size` aangemaakt. Met het idee om de grootte van een icon te kunnen instellen. In het geval van de Select vaak een chevron, maar als startpunt voor andere inputfields misschien een goed gebruik? Voor nu heb ik hem weggelaten.

Daarnaast leg ik hier de vraag neer of het gewenst is deze chevron custom te maken? Als dit een custom icoon zou kunnen worden moet deze sowieso mee kunnen kleuren met verschillende modes.
